openmc/filter.py

```diff
@@ -964,6 +965,66 @@ def get_pandas_dataframe(self, data_size, stride, **kwargs):
         return pd.concat([df, pd.DataFrame(filter_dict)])
 
 
+class CollisionFilter(Filter):
+    """Bins tally events based on the number of collisions.
+
+    Parameters
+    ----------
+    bins : Iterable of int
+        A list or iterable of the number of collisions, as integer values. 
+        The events whose post-scattering collision number equals one of 
+        the provided values will be counted. 
+    filter_id : int
+        Unique identifier for the filter
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the filter
+    bins : numpy.ndarray
+        An array of integer values representing the number of collisions events
+        by which to filter
+    num_bins : int
+        The number of filter bins
+
+    """
+
+    def __init__(self, bins, filter_id=None):
+        self.bins = np.asarray(bins)
+        self.id = filter_id
+
```

Comment:
```suggestion
```

Reply:
Hey @paulromano, I'll address these, thank you! Sorry for being so late, as I mentioned in another comment I'm having difficulty getting the openmc tests to even run/pass on any of my machines. I'll try the above pytest --update . command again, though

Reply:
Ah, ok, I know a lot of people have had problems with getting tests to match on macs. What I can do for now is update the test and send a pull request to your branch so we can keep this moving.